diff -ruN '--exclude=.git' oasis-base/Makefile oasis-linux/Makefile
--- oasis-base/Makefile	2026-04-12 20:48:06.000000000 +0000
+++ oasis-linux/Makefile	2026-05-26 06:37:51.070785415 +0000
@@ -538,8 +538,8 @@
 BINDGEN		= bindgen
 PAHOLE		= pahole
 RESOLVE_BTFIDS	= $(objtree)/tools/bpf/resolve_btfids/resolve_btfids
-LEX		= flex
-YACC		= bison
+LEX		= lex
+YACC		= yacc
 AWK		= awk
 INSTALLKERNEL  := installkernel
 PERL		= perl
diff -ruN '--exclude=.git' oasis-base/arch/x86/Makefile oasis-linux/arch/x86/Makefile
--- oasis-base/arch/x86/Makefile	2026-04-12 20:48:06.000000000 +0000
+++ oasis-linux/arch/x86/Makefile	2026-05-26 06:37:52.356909310 +0000
@@ -309,7 +309,7 @@
 endif
 	$(Q)$(MAKE) $(build)=$(boot) $(KBUILD_IMAGE)
 	$(Q)mkdir -p $(objtree)/arch/$(UTS_MACHINE)/boot
-	$(Q)ln -fsn ../../x86/boot/bzImage $(objtree)/arch/$(UTS_MACHINE)/boot/$@
+	$(Q)ln -fs ../../x86/boot/bzImage $(objtree)/arch/$(UTS_MACHINE)/boot/$@
 
 $(BOOT_TARGETS): vmlinux
 	$(Q)$(MAKE) $(build)=$(boot) $@
diff -ruN '--exclude=.git' oasis-base/certs/Makefile oasis-linux/certs/Makefile
--- oasis-base/certs/Makefile	2026-04-12 20:48:06.000000000 +0000
+++ oasis-linux/certs/Makefile	2026-05-26 06:37:52.508767251 +0000
@@ -87,5 +87,5 @@
 
 hostprogs := extract-cert
 
-HOSTCFLAGS_extract-cert.o = $(shell $(HOSTPKG_CONFIG) --cflags libcrypto 2> /dev/null) -I$(srctree)/scripts
-HOSTLDLIBS_extract-cert = $(shell $(HOSTPKG_CONFIG) --libs libcrypto 2> /dev/null || echo -lcrypto)
+HOSTCFLAGS_extract-cert.o =
+HOSTLDLIBS_extract-cert = -l bearssl
diff -ruN '--exclude=.git' oasis-base/certs/extract-cert.c oasis-linux/certs/extract-cert.c
--- oasis-base/certs/extract-cert.c	2026-04-12 20:48:06.000000000 +0000
+++ oasis-linux/certs/extract-cert.c	2026-05-26 06:37:52.508767251 +0000
@@ -18,20 +18,7 @@
 #include <stdbool.h>
 #include <string.h>
 #include <err.h>
-#include <openssl/bio.h>
-#include <openssl/pem.h>
-#include <openssl/err.h>
-#if OPENSSL_VERSION_MAJOR >= 3
-# define USE_PKCS11_PROVIDER
-# include <openssl/provider.h>
-# include <openssl/store.h>
-#else
-# if !defined(OPENSSL_NO_ENGINE) && !defined(OPENSSL_NO_DEPRECATED_3_0)
-#  define USE_PKCS11_ENGINE
-#  include <openssl/engine.h>
-# endif
-#endif
-#include "ssl-common.h"
+#include <bearssl.h>
 
 #define PKEY_ID_PKCS7 2
 
@@ -44,82 +31,15 @@
 }
 
 static const char *key_pass;
-static BIO *wb;
 static char *cert_dst;
 static bool verbose;
 
-static void write_cert(X509 *x509)
+static void write_cert(void *dst_ctx, const void *src, size_t len)
 {
-	char buf[200];
+	FILE *dst = dst_ctx;
 
-	if (!wb) {
-		wb = BIO_new_file(cert_dst, "wb");
-		ERR(!wb, "%s", cert_dst);
-	}
-	X509_NAME_oneline(X509_get_subject_name(x509), buf, sizeof(buf));
-	ERR(!i2d_X509_bio(wb, x509), "%s", cert_dst);
-	if (verbose)
-		fprintf(stderr, "Extracted cert: %s\n", buf);
-}
-
-static X509 *load_cert_pkcs11(const char *cert_src)
-{
-	X509 *cert = NULL;
-#ifdef USE_PKCS11_PROVIDER
-	OSSL_STORE_CTX *store;
-
-	if (!OSSL_PROVIDER_try_load(NULL, "pkcs11", true))
-		ERR(1, "OSSL_PROVIDER_try_load(pkcs11)");
-	if (!OSSL_PROVIDER_try_load(NULL, "default", true))
-		ERR(1, "OSSL_PROVIDER_try_load(default)");
-
-	store = OSSL_STORE_open(cert_src, NULL, NULL, NULL, NULL);
-	ERR(!store, "OSSL_STORE_open");
-
-	while (!OSSL_STORE_eof(store)) {
-		OSSL_STORE_INFO *info = OSSL_STORE_load(store);
-
-		if (!info) {
-			drain_openssl_errors(__LINE__, 0);
-			continue;
-		}
-		if (OSSL_STORE_INFO_get_type(info) == OSSL_STORE_INFO_CERT) {
-			cert = OSSL_STORE_INFO_get1_CERT(info);
-			ERR(!cert, "OSSL_STORE_INFO_get1_CERT");
-		}
-		OSSL_STORE_INFO_free(info);
-		if (cert)
-			break;
-	}
-	OSSL_STORE_close(store);
-#elif defined(USE_PKCS11_ENGINE)
-		ENGINE *e;
-		struct {
-			const char *cert_id;
-			X509 *cert;
-		} parms;
-
-		parms.cert_id = cert_src;
-		parms.cert = NULL;
-
-		ENGINE_load_builtin_engines();
-		drain_openssl_errors(__LINE__, 1);
-		e = ENGINE_by_id("pkcs11");
-		ERR(!e, "Load PKCS#11 ENGINE");
-		if (ENGINE_init(e))
-			drain_openssl_errors(__LINE__, 1);
-		else
-			ERR(1, "ENGINE_init");
-		if (key_pass)
-			ERR(!ENGINE_ctrl_cmd_string(e, "PIN", key_pass, 0), "Set PKCS#11 PIN");
-		ENGINE_ctrl_cmd(e, "LOAD_CERT_CTRL", 0, &parms, NULL, 1);
-		ERR(!parms.cert, "Get X.509 from PKCS#11");
-		cert = parms.cert;
-#else
-		fprintf(stderr, "no pkcs11 engine/provider available\n");
-		exit(1);
-#endif
-	return cert;
+	if (fwrite(src, 1, len, dst) != len)
+		err(1, "write %s", cert_dst);
 }
 
 int main(int argc, char **argv)
@@ -127,10 +47,6 @@
 	char *cert_src;
 	char *verbose_env;
 
-	OpenSSL_add_all_algorithms();
-	ERR_load_crypto_strings();
-	ERR_clear_error();
-
 	verbose_env = getenv("KBUILD_VERBOSE");
 	if (verbose_env && strchr(verbose_env, '1'))
 		verbose = true;
@@ -146,37 +62,45 @@
 	if (!cert_src[0]) {
 		/* Invoked with no input; create empty file */
 		FILE *f = fopen(cert_dst, "wb");
-		ERR(!f, "%s", cert_dst);
+		if (!f)
+			err(1, "%s", cert_dst);
 		fclose(f);
 		exit(0);
 	} else if (!strncmp(cert_src, "pkcs11:", 7)) {
-		X509 *cert = load_cert_pkcs11(cert_src);
-
-		ERR(!cert, "load_cert_pkcs11 failed");
-		write_cert(cert);
+		errx(1, "PKCS#11 certificates are not supported");
 	} else {
-		BIO *b;
-		X509 *x509;
-
-		b = BIO_new_file(cert_src, "rb");
-		ERR(!b, "%s", cert_src);
-
-		while (1) {
-			x509 = PEM_read_bio_X509(b, NULL, NULL, NULL);
-			if (wb && !x509) {
-				unsigned long err = ERR_peek_last_error();
-				if (ERR_GET_LIB(err) == ERR_LIB_PEM &&
-				    ERR_GET_REASON(err) == PEM_R_NO_START_LINE) {
-					ERR_clear_error();
+		FILE *src, *dst;
+		br_pem_decoder_context ctx;
+		char buf[8192], *pos;
+		size_t len = 0, n;
+
+		src = fopen(cert_src, "rb");
+		if (!src)
+			err(1, "open %s", cert_src);
+		dst = fopen(cert_dst, "wb");
+		if (!dst)
+			err(1, "open %s", cert_dst);
+
+		br_pem_decoder_init(&ctx);
+		br_pem_decoder_setdest(&ctx, write_cert, dst);
+		for (;;) {
+			if (len == 0) {
+				if (feof(src))
 					break;
-				}
+				len = fread(buf, 1, sizeof(buf), src);
+				if (ferror(src))
+					err(1, "read %s", cert_src);
+				pos = buf;
 			}
-			ERR(!x509, "%s", cert_src);
-			write_cert(x509);
+			n = br_pem_decoder_push(&ctx, pos, len);
+			pos += n;
+			len -= n;
+			if (br_pem_decoder_event(&ctx) == BR_PEM_ERROR)
+				errx(1, "PEM decode failure");
 		}
+		if (fflush(dst) != 0)
+			err(1, "flush %s", cert_dst);
 	}
 
-	BIO_free(wb);
-
 	return 0;
 }
diff -ruN '--exclude=.git' oasis-base/drivers/gpu/drm/xe/Makefile oasis-linux/drivers/gpu/drm/xe/Makefile
--- oasis-base/drivers/gpu/drm/xe/Makefile	2026-04-12 20:48:06.000000000 +0000
+++ oasis-linux/drivers/gpu/drm/xe/Makefile	2026-05-26 06:37:55.094072116 +0000
@@ -361,9 +361,9 @@
 obj-$(CONFIG_DRM_XE_KUNIT_TEST) += tests/
 
 # header test
-hdrtest_find_args := -not -path xe_rtp_helpers.h
+hdrtest_find_args := ! -path xe_rtp_helpers.h
 ifneq ($(CONFIG_DRM_XE_DISPLAY),y)
-	hdrtest_find_args += -not -path display/\* -not -path compat-i915-headers/\* -not -path xe_display.h
+	hdrtest_find_args += ! -path display/\* ! -path compat-i915-headers/\* ! -path xe_display.h
 endif
 
 always-$(CONFIG_DRM_XE_WERROR) += \
diff -ruN '--exclude=.git' oasis-base/lib/Makefile oasis-linux/lib/Makefile
--- oasis-base/lib/Makefile	2026-04-12 20:48:06.000000000 +0000
+++ oasis-linux/lib/Makefile	2026-05-26 06:37:59.234982879 +0000
@@ -293,11 +293,11 @@
 $(obj)/oid_registry.o: $(obj)/oid_registry_data.c
 
 $(obj)/oid_registry_data.c: $(srctree)/include/linux/oid_registry.h \
-			    $(src)/build_OID_registry
+			    $(src)/build_OID_registry.awk
 	$(call cmd,build_OID_registry)
 
 quiet_cmd_build_OID_registry = GEN     $@
-      cmd_build_OID_registry = perl $(src)/build_OID_registry $< $@
+      cmd_build_OID_registry = awk -f $(srctree)/$(src)/build_OID_registry.awk $< > $@
 
 clean-files	+= oid_registry_data.c
 
diff -ruN '--exclude=.git' oasis-base/lib/build_OID_registry oasis-linux/lib/build_OID_registry
--- oasis-base/lib/build_OID_registry	2026-04-12 20:48:06.000000000 +0000
+++ oasis-linux/lib/build_OID_registry	1970-01-01 00:00:00.000000000 +0000
@@ -1,218 +0,0 @@
-#!/usr/bin/perl -w
-# SPDX-License-Identifier: GPL-2.0-or-later
-#
-# Build a static ASN.1 Object Identified (OID) registry
-#
-# Copyright (C) 2012 Red Hat, Inc. All Rights Reserved.
-# Written by David Howells (dhowells@redhat.com)
-#
-
-use strict;
-use Cwd qw(abs_path);
-
-my @names = ();
-my @oids = ();
-
-if ($#ARGV != 1) {
-    print STDERR "Format: ", $0, " <in-h-file> <out-c-file>\n";
-    exit(2);
-}
-
-my $abs_srctree = abs_path($ENV{'srctree'});
-
-#
-# Open the file to read from
-#
-open IN_FILE, "<$ARGV[0]" || die;
-while (<IN_FILE>) {
-    chomp;
-    if (m!\s+OID_([a-zA-z][a-zA-Z0-9_]+),\s+/[*]\s+([012][.0-9]*)\s+[*]/!) {
-	push @names, $1;
-	push @oids, $2;
-    }
-}
-close IN_FILE || die;
-
-#
-# Open the files to write into
-#
-open C_FILE, ">$ARGV[1]" or die;
-print C_FILE "/*\n";
-my $scriptname = $0;
-$scriptname =~ s#^\Q$abs_srctree/\E##;
-print C_FILE " * Automatically generated by ", $scriptname, ".  Do not edit\n";
-print C_FILE " */\n";
-
-#
-# Split the data up into separate lists and also determine the lengths of the
-# encoded data arrays.
-#
-my @indices = ();
-my @lengths = ();
-my $total_length = 0;
-
-for (my $i = 0; $i <= $#names; $i++) {
-    my $name = $names[$i];
-    my $oid = $oids[$i];
-
-    my @components = split(/[.]/, $oid);
-
-    # Determine the encoded length of this OID
-    my $size = $#components;
-    for (my $loop = 2; $loop <= $#components; $loop++) {
-	$ENV{'BC_LINE_LENGTH'} = "0";
-	my $c = `echo "ibase=10; obase=2; $components[$loop]" | bc`;
-	chomp($c);
-
-	# We will base128 encode the number
-	my $tmp = length($c) - 1;
-	$tmp = int($tmp / 7);
-	$size += $tmp;
-    }
-    push @lengths, $size;
-    push @indices, $total_length;
-    $total_length += $size;
-}
-
-#
-# Emit the look-up-by-OID index table
-#
-print C_FILE "\n";
-if ($total_length <= 255) {
-    print C_FILE "static const unsigned char oid_index[OID__NR + 1] = {\n";
-} else {
-    print C_FILE "static const unsigned short oid_index[OID__NR + 1] = {\n";
-}
-for (my $i = 0; $i <= $#names; $i++) {
-    print C_FILE "\t[OID_", $names[$i], "] = ", $indices[$i], ",\n"
-}
-print C_FILE "\t[OID__NR] = ", $total_length, "\n";
-print C_FILE "};\n";
-
-#
-# Encode the OIDs
-#
-my @encoded_oids = ();
-
-for (my $i = 0; $i <= $#names; $i++) {
-    my @octets = ();
-
-    my @components = split(/[.]/, $oids[$i]);
-
-    push @octets, $components[0] * 40 + $components[1];
-
-    for (my $loop = 2; $loop <= $#components; $loop++) {
-	# get the base 2 representation of the component
-	$ENV{'BC_LINE_LENGTH'} = "0";
-	my $c = `echo "ibase=10; obase=2; $components[$loop]" | bc`;
-	chomp($c);
-
-	my $tmp = length($c) - 1;
-	$tmp = int($tmp / 7);
-
-	# zero pad upto length multiple of 7
-	$c = substr("0000000", 0, ($tmp + 1) * 7 - length($c)).$c;
-
-	# Base128 encode the number
-	for (my $j = 0; $j < $tmp; $j++) {
-	    my $b = oct("0b".substr($c, $j * 7, 7));
-
-	    push @octets, $b | 0x80;
-	}
-	push @octets, oct("0b".substr($c, $tmp * 7, 7));
-    }
-
-    push @encoded_oids, \@octets;
-}
-
-#
-# Create a hash value for each OID
-#
-my @hash_values = ();
-for (my $i = 0; $i <= $#names; $i++) {
-    my @octets = @{$encoded_oids[$i]};
-
-    my $hash = $#octets;
-    foreach (@octets) {
-	$hash += $_ * 33;
-    }
-
-    $hash = ($hash >> 24) ^ ($hash >> 16) ^ ($hash >> 8) ^ ($hash);
-
-    push @hash_values, $hash & 0xff;
-}
-
-#
-# Emit the OID data
-#
-print C_FILE "\n";
-print C_FILE "static const unsigned char oid_data[", $total_length, "] = {\n";
-for (my $i = 0; $i <= $#names; $i++) {
-    my @octets = @{$encoded_oids[$i]};
-    print C_FILE "\t";
-    print C_FILE $_, ", " foreach (@octets);
-    print C_FILE "\t// ", $names[$i];
-    print C_FILE "\n";
-}
-print C_FILE "};\n";
-
-#
-# Build the search index table (ordered by length then hash then content)
-#
-my @index_table = ( 0 .. $#names );
-
-@index_table = sort {
-    my @octets_a = @{$encoded_oids[$a]};
-    my @octets_b = @{$encoded_oids[$b]};
-
-    return $hash_values[$a] <=> $hash_values[$b]
-	if ($hash_values[$a] != $hash_values[$b]);
-    return $#octets_a <=> $#octets_b
-	if ($#octets_a != $#octets_b);
-    for (my $i = $#octets_a; $i >= 0; $i--) {
-	return $octets_a[$i] <=> $octets_b[$i]
-	    if ($octets_a[$i] != $octets_b[$i]);
-    }
-    return 0;
-
-} @index_table;
-
-#
-# Emit the search index and hash value table
-#
-print C_FILE "\n";
-print C_FILE "static const struct {\n";
-print C_FILE "\tunsigned char hash;\n";
-if ($#names <= 255) {
-    print C_FILE "\tenum OID oid : 8;\n";
-} else {
-    print C_FILE "\tenum OID oid : 16;\n";
-}
-print C_FILE "} oid_search_table[OID__NR] = {\n";
-for (my $i = 0; $i <= $#names; $i++) {
-    my @octets = @{$encoded_oids[$index_table[$i]]};
-    printf(C_FILE "\t[%3u] = { %3u, OID_%-35s }, // ",
-	   $i,
-	   $hash_values[$index_table[$i]],
-	   $names[$index_table[$i]]);
-    printf C_FILE "%02x", $_ foreach (@octets);
-    print C_FILE "\n";
-}
-print C_FILE "};\n";
-
-#
-# Emit the OID debugging name table
-#
-#print C_FILE "\n";
-#print C_FILE "const char *const oid_name_table[OID__NR + 1] = {\n";
-#
-#for (my $i = 0; $i <= $#names; $i++) {
-#    print C_FILE "\t\"", $names[$i], "\",\n"
-#}
-#print C_FILE "\t\"Unknown-OID\"\n";
-#print C_FILE "};\n";
-
-#
-# Polish off
-#
-close C_FILE or die;
diff -ruN '--exclude=.git' oasis-base/lib/build_OID_registry.awk oasis-linux/lib/build_OID_registry.awk
--- oasis-base/lib/build_OID_registry.awk	1970-01-01 00:00:00.000000000 +0000
+++ oasis-linux/lib/build_OID_registry.awk	2026-05-26 06:37:59.236982927 +0000
@@ -0,0 +1,144 @@
+# SPDX-License-Identifier: GPL-2.0-or-later
+#
+# Build a static ASN.1 Object Identified (OID) registry
+#
+# Copyright (C) 2012 Red Hat, Inc. All Rights Reserved.
+# Based on perl script written by David Howells (dhowells@redhat.com)
+
+function bit(n, i) {
+	return n % 2 ^ (i + 1) > 2 ^ i - 1
+}
+
+function divmod(d, q,	x, n, i, j) {
+	x = 0
+	n = d[0]
+	for (i = 1; i <= n && x < q; ++i)
+		x = x * 10 + d[i]
+	j = 0
+	for (;; ++i) {
+		d[++j] = int(x / q)
+		x %= q
+		if (i > n)
+			break
+		x = x * 10 + d[i]
+	}
+	d[0] = j
+	return x
+}
+
+#
+# Read OID lines and determine the lengths of the encoded data arrays.
+#
+/[ \t]+OID_[a-zA-Z][a-zA-Z0-9_]+,[ \t]+\/\*[ \t]+[012][.0-9]*[ \t]+\*\// {
+	++num_oids
+
+	names[num_oids] = substr($1, 5, length($1) - 5)
+	oids[num_oids] = $3
+	indices[num_oids] = total_length
+	sorted[num_oids] = num_oids
+
+	# Encode the OID
+	n = 1
+	num_components = split($3, components, ".")
+	octets[num_oids, n] = components[1] * 40 + components[2]
+	for (i = 3; i <= num_components; ++i) {
+		c = components[i]
+		for (j = 1; j <= length(c); ++j)
+			digits[j] = substr(c, j, 1)
+		digits[0] = j - 1
+
+		# Base128 encode the number
+		m = 0
+		while (digits[1] != 0)
+			c_base128[++m] = divmod(digits, 128)
+
+		n += m
+		octets[num_oids, n] = c_base128[1]
+		for (j = 1; j < m; ++j)
+			octets[num_oids, n - j] = c_base128[j + 1] + 128
+	}
+	total_length += n
+
+	# Calculate the hash
+	h = n - 1
+	for (i = 1; i <= n; ++i)
+		h += octets[num_oids, i] * 33
+	hash = 0
+	for (i = 7; i >= 0; --i)
+		hash = hash * 2 + (((bit(h, i) != bit(h, i + 8)) != bit(h, i + 16)) != bit(h, i + 24))
+
+	num_octets[num_oids] = n
+	hashes[num_oids] = hash
+}
+
+function isless(i, j) {
+	if (hashes[i] != hashes[j])
+		return hashes[i] < hashes[j]
+	if (num_octets[i] != num_octets[j])
+		return num_octets[i] < num_octets[j]
+	for (k = num_octets[i]; k > 0; --k) {
+		if (octets[i, k] != octets[j, k])
+			return octets[i, k] < octets[j, k]
+	}
+	return 0
+}
+
+function sort(lo, hi) {
+	if (lo >= hi)
+		return
+	pivot = sorted[hi]
+	i = lo
+	for (j = lo; j <= hi; ++j) {
+		if (!isless(sorted[j], pivot))
+			continue
+		tmp = sorted[i]
+		sorted[i] = sorted[j]
+		sorted[j] = tmp
+		++i
+	}
+	tmp = sorted[i]
+	sorted[i] = sorted[hi]
+	sorted[hi] = tmp
+	sort(lo, i - 1)
+	sort(i + 1, hi)
+}
+
+END {
+	#
+	# Emit the look-up-by-OID index table
+	#
+	printf "/*\n * Automatically generated by lib/build_OID_registry.awk.  Do not edit\n */\n\n"
+	printf "static const unsigned %s oid_index[OID__NR + 1] = {\n", (total_length <= 255) ? "char" : "short"
+	for (i = 1; i <= num_oids; ++i)
+		printf "\t[OID_%s] = %u,\n", names[i], indices[i]
+	printf "\t[OID__NR] = %u\n};\n\n", total_length
+
+	#
+	# Emit the OID data
+	#
+	printf "static const unsigned char oid_data[%u] = {\n", total_length
+	for (i = 1; i <= num_oids; ++i) {
+		printf "\t"
+		for (j = 1; j <= num_octets[i]; ++j)
+			printf "%u, ", octets[i, j]
+		printf "\t// %s\n", names[i]
+	}
+	printf "};\n\n"
+
+	#
+	# Emit the search index and hash value table
+	#
+	sort(1, num_oids)
+	printf "static const struct {\n"
+	printf "\tunsigned char hash;\n"
+	printf "\tenum OID oid : %u;\n", (num_oids <= 256) ? 8 : 16
+	printf "} oid_search_table[OID__NR] = {\n"
+	for (i = 1; i <= num_oids; ++i) {
+		j = sorted[i]
+		printf "\t[%3u] = { %3u, OID_%-35s }, // ", i - 1, hashes[j], names[j]
+		for (k = 1; k <= num_octets[j]; ++k)
+			printf "%02x", octets[j, k]
+		printf "\n"
+	}
+	printf "};\n"
+}
diff -ruN '--exclude=.git' oasis-base/net/9p/error.c oasis-linux/net/9p/error.c
--- oasis-base/net/9p/error.c	2026-04-12 20:48:06.000000000 +0000
+++ oasis-linux/net/9p/error.c	2026-05-26 06:37:59.339979685 +0000
@@ -163,6 +163,7 @@
 	{"cannot remove root", EPERM},
 	{"file too big", EFBIG},
 	{"venti i/o error", EIO},
+	{"does not exist", ENOENT},
 	/* these are not errors */
 	{"u9fs rhostsauth: no authentication required", 0},
 	{"u9fs authnone: no authentication required", 0},
@@ -203,6 +204,21 @@
 	int errno;
 	struct errormap *c;
 	u32 hash;
+	char *p;
+
+	/* strip possible quoted filename */
+	errstr[len] = 0;
+	p = errstr;
+	if (*p == '\'') {
+		while (*++p) {
+			if (*p == '\'' && *++p != '\'')
+				break;
+		}
+		if (*p == ' ')
+			p++;
+	}
+	len -= p - errstr;
+	errstr = p;
 
 	errno = 0;
 	c = NULL;
@@ -216,7 +232,6 @@
 
 	if (errno == 0) {
 		/* TODO: if error isn't found, add it dynamically */
-		errstr[len] = 0;
 		pr_err("%s: server reported unknown error %s\n",
 		       __func__, errstr);
 		errno = ESERVERFAULT;
diff -ruN '--exclude=.git' oasis-base/scripts/Makefile.host oasis-linux/scripts/Makefile.host
--- oasis-base/scripts/Makefile.host	2026-04-12 20:48:06.000000000 +0000
+++ oasis-linux/scripts/Makefile.host	2026-05-26 06:37:59.596991572 +0000
@@ -11,7 +11,7 @@
 # YACC
 # ---------------------------------------------------------------------------
 quiet_cmd_bison = YACC    $(basename $@).[ch]
-      cmd_bison = $(YACC) -o $(basename $@).c --defines=$(basename $@).h -t -l $<
+      cmd_bison = $(YACC) -o $(basename $@).c -d -t -l $<
 
 $(obj)/%.tab.c $(obj)/%.tab.h: $(src)/%.y FORCE
 	$(call if_changed,bison)
diff -ruN '--exclude=.git' oasis-base/scripts/kconfig/Makefile oasis-linux/scripts/kconfig/Makefile
--- oasis-base/scripts/kconfig/Makefile	2026-04-12 20:48:06.000000000 +0000
+++ oasis-linux/scripts/kconfig/Makefile	2026-05-26 06:37:59.621548794 +0000
@@ -166,7 +166,7 @@
 common-objs	:= confdata.o expr.o lexer.lex.o menu.o parser.tab.o \
 		   preprocess.o symbol.o util.o
 
-$(obj)/lexer.lex.o: $(obj)/parser.tab.h
+$(obj)/lexer.lex.o: $(obj)/parser.tab.c
 HOSTCFLAGS_lexer.lex.o	:= -I $(src)
 HOSTCFLAGS_parser.tab.o	:= -I $(src)
 
diff -ruN '--exclude=.git' oasis-base/scripts/kconfig/lxdialog/dialog.h oasis-linux/scripts/kconfig/lxdialog/dialog.h
--- oasis-base/scripts/kconfig/lxdialog/dialog.h	2026-04-12 20:48:06.000000000 +0000
+++ oasis-linux/scripts/kconfig/lxdialog/dialog.h	2026-05-26 06:37:59.623379830 +0000
@@ -16,7 +16,7 @@
 #ifdef __sun__
 #define CURS_MACROS
 #endif
-#include <ncurses.h>
+#include <curses.h>
 
 #define TR(params) _tracef params
 
diff -ruN '--exclude=.git' oasis-base/scripts/kconfig/mconf-cfg.sh oasis-linux/scripts/kconfig/mconf-cfg.sh
--- oasis-base/scripts/kconfig/mconf-cfg.sh	2026-04-12 20:48:06.000000000 +0000
+++ oasis-linux/scripts/kconfig/mconf-cfg.sh	2026-05-26 06:37:59.623379830 +0000
@@ -6,52 +6,6 @@
 cflags=$1
 libs=$2
 
-PKG="ncursesw"
-PKG2="ncurses"
-
-if [ -n "$(command -v ${HOSTPKG_CONFIG})" ]; then
-	if ${HOSTPKG_CONFIG} --exists $PKG; then
-		${HOSTPKG_CONFIG} --cflags ${PKG} > ${cflags}
-		${HOSTPKG_CONFIG} --libs ${PKG} > ${libs}
-		exit 0
-	fi
-
-	if ${HOSTPKG_CONFIG} --exists ${PKG2}; then
-		${HOSTPKG_CONFIG} --cflags ${PKG2} > ${cflags}
-		${HOSTPKG_CONFIG} --libs ${PKG2} > ${libs}
-		exit 0
-	fi
-fi
-
-# Check the default paths in case pkg-config is not installed.
-# (Even if it is installed, some distributions such as openSUSE cannot
-# find ncurses by pkg-config.)
-if [ -f /usr/include/ncursesw/ncurses.h ]; then
-	echo -D_GNU_SOURCE -I/usr/include/ncursesw > ${cflags}
-	echo -lncursesw > ${libs}
-	exit 0
-fi
-
-if [ -f /usr/include/ncurses/ncurses.h ]; then
-	echo -D_GNU_SOURCE -I/usr/include/ncurses > ${cflags}
-	echo -lncurses > ${libs}
-	exit 0
-fi
-
-# As a final fallback before giving up, check if $HOSTCC knows of a default
-# ncurses installation (e.g. from a vendor-specific sysroot).
-if echo '#include <ncurses.h>' | ${HOSTCC} -E - >/dev/null 2>&1; then
-	echo -D_GNU_SOURCE > ${cflags}
-	echo -lncurses > ${libs}
-	exit 0
-fi
-
-echo >&2 "*"
-echo >&2 "* Unable to find the ncurses package."
-echo >&2 "* Install ncurses (ncurses-devel or libncurses-dev"
-echo >&2 "* depending on your distribution)."
-echo >&2 "*"
-echo >&2 "* You may also need to install ${HOSTPKG_CONFIG} to find the"
-echo >&2 "* ncurses installed in a non-default location."
-echo >&2 "*"
-exit 1
+echo -D_GNU_SOURCE > ${cflags}
+echo -lcurses -lterminfo > ${libs}
+exit 0
diff -ruN '--exclude=.git' oasis-base/scripts/kconfig/parser.y oasis-linux/scripts/kconfig/parser.y
--- oasis-base/scripts/kconfig/parser.y	2026-04-12 20:48:06.000000000 +0000
+++ oasis-linux/scripts/kconfig/parser.y	2026-05-26 06:37:59.623992220 +0000
@@ -97,13 +97,6 @@
 %type <string> assign_val
 %type <flavor> assign_op
 
-%destructor {
-	fprintf(stderr, "%s:%d: missing end statement for this entry\n",
-		$$->filename, $$->lineno);
-	if (current_menu == $$)
-		menu_end_menu();
-} if_entry menu_entry choice_entry
-
 %%
 input: mainmenu_stmt stmt_list | stmt_list;
 
diff -ruN '--exclude=.git' oasis-base/tools/build/Makefile oasis-linux/tools/build/Makefile
--- oasis-base/tools/build/Makefile	2026-04-12 20:48:06.000000000 +0000
+++ oasis-linux/tools/build/Makefile	2026-05-26 06:37:59.992193533 +0000
@@ -30,7 +30,7 @@
 
 clean:
 	$(call QUIET_CLEAN, fixdep)
-	$(Q)find $(or $(OUTPUT),.) -name '*.o' -delete -o -name '\.*.cmd' -delete -o -name '\.*.d' -delete
+	$(Q)find $(or $(OUTPUT),.) '(' -name '*.o' -o -name '\.*.cmd' -o -name '\.*.d' ')' -exec $(RM) {} +
 	$(Q)rm -f $(OUTPUT)fixdep
 	$(call QUIET_CLEAN, feature-detect)
 ifneq ($(wildcard $(TMP_O)),)
diff -ruN '--exclude=.git' oasis-base/tools/build/tests/ex/Makefile oasis-linux/tools/build/tests/ex/Makefile
--- oasis-base/tools/build/tests/ex/Makefile	2026-04-12 20:48:06.000000000 +0000
+++ oasis-linux/tools/build/tests/ex/Makefile	2026-05-26 06:37:59.995873812 +0000
@@ -21,7 +21,7 @@
 	make $(build)=libex
 
 clean:
-	find . -name '*.o' -delete -o -name '\.*.cmd' -delete -o -name '\.*.d' -delete
+	find . '(' -name '*.o' -o -name '\.*.cmd' -o -name '\.*.d' ')' -exec $(RM) {} +
 	rm -f ex ex.i ex.s
 
 .PHONY: FORCE
diff -ruN '--exclude=.git' oasis-base/tools/gpio/Makefile oasis-linux/tools/gpio/Makefile
--- oasis-base/tools/gpio/Makefile	2026-04-12 20:48:06.000000000 +0000
+++ oasis-linux/tools/gpio/Makefile	2026-05-26 06:37:59.999989046 +0000
@@ -77,8 +77,8 @@
 
 clean:
 	rm -f $(ALL_PROGRAMS)
-	rm -rf $(OUTPUT)include
-	find $(or $(OUTPUT),.) -name '*.o' -delete -o -name '\.*.d' -delete -o -name '\.*.cmd' -delete
+	rm -f $(OUTPUT)include/linux/gpio.h
+	find $(or $(OUTPUT),.) '(' -name '*.o' -o -name '\.*.d' -o -name '\.*.cmd' ')' -exec $(RM) {} +
 
 install: $(ALL_PROGRAMS)
 	install -d -m 755 $(DESTDIR)$(bindir);		\
diff -ruN '--exclude=.git' oasis-base/tools/hv/Makefile oasis-linux/tools/hv/Makefile
--- oasis-base/tools/hv/Makefile	2026-04-12 20:48:06.000000000 +0000
+++ oasis-linux/tools/hv/Makefile	2026-05-26 06:38:00.000601518 +0000
@@ -52,7 +52,7 @@
 
 clean:
 	rm -f $(ALL_PROGRAMS)
-	find $(or $(OUTPUT),.) -name '*.o' -delete -o -name '\.*.d' -delete -o -name '\.*.cmd' -delete
+	find $(or $(OUTPUT),.) '(' -name '*.o' -o -name '\.*.d' -o -name '\.*.cmd' ')' -exec $(RM) {} +
 
 install: $(ALL_PROGRAMS)
 	install -d -m 755 $(DESTDIR)$(sbindir); \
diff -ruN '--exclude=.git' oasis-base/tools/iio/Makefile oasis-linux/tools/iio/Makefile
--- oasis-base/tools/iio/Makefile	2026-04-12 20:48:06.000000000 +0000
+++ oasis-linux/tools/iio/Makefile	2026-05-26 06:38:00.001591788 +0000
@@ -58,7 +58,7 @@
 clean:
 	rm -f $(ALL_PROGRAMS)
 	rm -rf $(OUTPUT)include/linux/iio
-	find $(or $(OUTPUT),.) -name '*.o' -delete -o -name '\.*.d' -delete -o -name '\.*.cmd' -delete
+	find $(or $(OUTPUT),.) '(' -name '*.o' -o -name '\.*.d' -o -name '\.*.cmd' ')' -exec $(RM) {} +
 
 install: $(ALL_PROGRAMS)
 	install -d -m 755 $(DESTDIR)$(bindir);		\
diff -ruN '--exclude=.git' oasis-base/tools/lib/api/Makefile oasis-linux/tools/lib/api/Makefile
--- oasis-base/tools/lib/api/Makefile	2026-04-12 20:48:06.000000000 +0000
+++ oasis-linux/tools/lib/api/Makefile	2026-05-26 06:38:01.389511000 +0000
@@ -87,7 +87,7 @@
 	if [ ! -d '$2' ]; then             \
 		$(INSTALL) -d -m 755 '$2'; \
 	fi;                                \
-	$(INSTALL) $1 $(if $3,-m $3,) '$2'
+	$(INSTALL) $(if $3,-m $3,) $1 '$2'
 endef
 
 install_lib: $(LIBFILE)
@@ -122,7 +122,7 @@
 
 clean:
 	$(call QUIET_CLEAN, libapi) $(RM) $(LIBFILE); \
-	find $(or $(OUTPUT),.) -name \*.o -or -name \*.o.cmd -or -name \*.o.d | xargs $(RM)
+	find $(or $(OUTPUT),.) '(' -name \*.o -o -name \*.o.cmd -o -name \*.o.d ')' -exec $(RM) {} +
 
 FORCE:
 
diff -ruN '--exclude=.git' oasis-base/tools/lib/perf/Makefile oasis-linux/tools/lib/perf/Makefile
--- oasis-base/tools/lib/perf/Makefile	2026-04-12 20:48:06.000000000 +0000
+++ oasis-linux/tools/lib/perf/Makefile	2026-05-26 06:38:01.397739976 +0000
@@ -160,7 +160,7 @@
 	if [ ! -d '$2' ]; then             \
 		$(INSTALL) -d -m 755 '$2'; \
 	fi;                                \
-	$(INSTALL) $1 $(if $3,-m $3,) '$2'
+	$(INSTALL) $(if $3,-m $3,) $1 '$2'
 endef
 
 install_lib: libs
diff -ruN '--exclude=.git' oasis-base/tools/lib/subcmd/Makefile oasis-linux/tools/lib/subcmd/Makefile
--- oasis-base/tools/lib/subcmd/Makefile	2026-04-12 20:48:06.000000000 +0000
+++ oasis-linux/tools/lib/subcmd/Makefile	2026-05-26 06:38:01.402485658 +0000
@@ -90,7 +90,7 @@
 	if [ ! -d '$2' ]; then             \
 		$(INSTALL) -d -m 755 '$2'; \
 	fi;                                             \
-	$(INSTALL) $1 $(if $3,-m $3,) '$2'
+	$(INSTALL) $(if $3,-m $3,) $1 '$2'
 endef
 
 install_lib: $(LIBFILE)
@@ -113,7 +113,7 @@
 
 clean:
 	$(call QUIET_CLEAN, libsubcmd) $(RM) $(LIBFILE); \
-	find $(or $(OUTPUT),.) -name \*.o -or -name \*.o.cmd -or -name \*.o.d | xargs $(RM)
+	find $(or $(OUTPUT),.) '(' -name \*.o -o -name \*.o.cmd -o -name \*.o.d ')' -exec $(RM) {} +
 
 FORCE:
 
diff -ruN '--exclude=.git' oasis-base/tools/lib/symbol/Makefile oasis-linux/tools/lib/symbol/Makefile
--- oasis-base/tools/lib/symbol/Makefile	2026-04-12 20:48:06.000000000 +0000
+++ oasis-linux/tools/lib/symbol/Makefile	2026-05-26 06:38:01.403458563 +0000
@@ -88,7 +88,7 @@
 	if [ ! -d '$2' ]; then             \
 		$(INSTALL) -d -m 755 '$2'; \
 	fi;                                \
-	$(INSTALL) $1 $(if $3,-m $3,) '$2'
+	$(INSTALL) $(if $3,-m $3,) $1 '$2'
 endef
 
 install_lib: $(LIBFILE)
diff -ruN '--exclude=.git' oasis-base/tools/objtool/Makefile oasis-linux/tools/objtool/Makefile
--- oasis-base/tools/objtool/Makefile	2026-04-12 20:48:06.000000000 +0000
+++ oasis-linux/tools/objtool/Makefile	2026-05-26 06:38:01.423545386 +0000
@@ -42,8 +42,8 @@
 OBJTOOL    := $(OUTPUT)objtool
 OBJTOOL_IN := $(OBJTOOL)-in.o
 
-LIBELF_FLAGS := $(shell $(HOSTPKG_CONFIG) libelf --cflags 2>/dev/null)
-LIBELF_LIBS  := $(shell $(HOSTPKG_CONFIG) libelf --libs 2>/dev/null || echo -lelf)
+LIBELF_FLAGS :=
+LIBELF_LIBS  := -l elf
 
 all: $(OBJTOOL)
 
@@ -142,7 +142,7 @@
 	$(Q)$(RM) -r -- $(LIBSUBCMD_OUTPUT)
 
 clean: $(LIBSUBCMD)-clean
-	$(Q)find $(OUTPUT) \( -name '*.o' -o -name '\.*.cmd' -o -name '\.*.d' \) -type f -print | xargs $(RM)
+	$(Q)find $(OUTPUT) '(' -name '*.o' -o -name '\.*.cmd' -o -name '\.*.d' ')' -exec $(RM) {} +
 	$(Q)$(RM) $(OUTPUT)arch/x86/lib/cpu-feature-names.c $(OUTPUT)fixdep
 	$(Q)$(RM) $(OUTPUT)arch/x86/lib/inat-tables.c $(OUTPUT)fixdep
 	$(Q)$(RM) -- $(OUTPUT)FEATURE-DUMP.objtool
diff -ruN '--exclude=.git' oasis-base/tools/objtool/elf.c oasis-linux/tools/objtool/elf.c
--- oasis-base/tools/objtool/elf.c	2026-04-12 20:48:06.000000000 +0000
+++ oasis-linux/tools/objtool/elf.c	2026-05-26 06:38:01.425037953 +0000
@@ -849,9 +849,16 @@
 		}
 	}
 
-	if (!gelf_update_symshndx(symtab_data, shndx_data, idx, &sym->sym, shndx)) {
-		ERROR_ELF("gelf_update_symshndx");
-		return -1;
+	if (shndx_data) {
+		if (!gelf_update_symshndx(symtab_data, shndx_data, idx, &sym->sym, shndx)) {
+			ERROR_ELF("gelf_update_symshndx");
+			return -1;
+		}
+	} else {
+		if (!gelf_update_sym(symtab_data, idx, &sym->sym)) {
+			ERROR_ELF("gelf_update_sym");
+			return -1;
+		}
 	}
 
 	return 0;
diff -ruN '--exclude=.git' oasis-base/tools/perf/Makefile.perf oasis-linux/tools/perf/Makefile.perf
--- oasis-base/tools/perf/Makefile.perf	2026-04-12 20:48:06.000000000 +0000
+++ oasis-linux/tools/perf/Makefile.perf	2026-05-26 06:38:01.434035685 +0000
@@ -1289,8 +1289,8 @@
 		tests-coresight-targets-clean pmu-events-clean
 	$(call QUIET_CLEAN, core-objs)  $(RM) $(LIBPERF_A) $(OUTPUT)perf-archive \
 		$(OUTPUT)perf-iostat $(LANG_BINDINGS)
-	$(Q)find $(or $(OUTPUT),.) -name '*.o' -delete -o -name '*.a' -delete -o \
-		-name '\.*.cmd' -delete -o -name '\.*.d' -delete -o -name '*.shellcheck_log' -delete
+	$(Q)find $(or $(OUTPUT),.) '(' -name '*.o' -o -name '*.a' -o \
+		-name '\.*.cmd' -o -name '\.*.d' -o -name '*.shellcheck_log' ')' -exec $(RM) {} +
 	$(Q)$(RM) $(OUTPUT).config-detected
 	$(call QUIET_CLEAN, core-progs) $(RM) $(ALL_PROGRAMS) perf perf-read-vdso32 \
 		perf-read-vdsox32 $(OUTPUT)$(LIBJVMTI).so
diff -ruN '--exclude=.git' oasis-base/tools/perf/trace/beauty/drm_ioctl.sh oasis-linux/tools/perf/trace/beauty/drm_ioctl.sh
--- oasis-base/tools/perf/trace/beauty/drm_ioctl.sh	2026-04-12 20:48:06.000000000 +0000
+++ oasis-linux/tools/perf/trace/beauty/drm_ioctl.sh	2026-05-26 06:38:01.576051017 +0000
@@ -9,7 +9,7 @@
 
 printf "static const char *drm_ioctl_cmds[] = {\n"
 grep "^#define DRM_IOCTL.*DRM_IO" $header_dir/drm.h | \
-	sed -r 's/^#define +DRM_IOCTL_([A-Z0-9_]+)[	 ]+DRM_IO[A-Z]* *\( *(0x[[:xdigit:]]+),*.*/	[\2] = "\1",/g'
+	sed -E 's/^#define +DRM_IOCTL_([A-Z0-9_]+)[	 ]+DRM_IO[A-Z]* *\( *(0x[[:xdigit:]]+),*.*/	[\2] = "\1",/g'
 grep "^#define DRM_I915_[A-Z_0-9]\+[	 ]\+0x" $header_dir/i915_drm.h | \
-	sed -r 's/^#define +DRM_I915_([A-Z0-9_]+)[	 ]+(0x[[:xdigit:]]+)/\t[DRM_COMMAND_BASE + \2] = "I915_\1",/g'
+	sed -E 's/^#define +DRM_I915_([A-Z0-9_]+)[	 ]+(0x[[:xdigit:]]+)/	[DRM_COMMAND_BASE + \2] = "I915_\1",/g'
 printf "};\n"
diff -ruN '--exclude=.git' oasis-base/tools/perf/trace/beauty/sndrv_ctl_ioctl.sh oasis-linux/tools/perf/trace/beauty/sndrv_ctl_ioctl.sh
--- oasis-base/tools/perf/trace/beauty/sndrv_ctl_ioctl.sh	2026-04-12 20:48:06.000000000 +0000
+++ oasis-linux/tools/perf/trace/beauty/sndrv_ctl_ioctl.sh	2026-05-26 06:38:01.577045314 +0000
@@ -4,6 +4,6 @@
 [ $# -eq 1 ] && beauty_uapi_sound_dir=$1 || beauty_uapi_sound_dir=tools/perf/trace/beauty/include/uapi/sound/
 
 printf "static const char *sndrv_ctl_ioctl_cmds[] = {\n"
-grep "^#define[\t ]\+SNDRV_CTL_IOCTL_" $beauty_uapi_sound_dir/asound.h | \
-	sed -r 's/^#define +SNDRV_CTL_IOCTL_([A-Z0-9_]+)[\t ]+_IO[RW]*\( *.U., *(0x[[:xdigit:]]+),?.*/\t[\2] = \"\1\",/g'
+grep "^#define[	 ]\+SNDRV_CTL_IOCTL_" $beauty_uapi_sound_dir/asound.h | \
+	sed -E 's/^#define +SNDRV_CTL_IOCTL_([A-Z0-9_]+)[	 ]+_IO[RW]*\( *.U., *(0x[[:xdigit:]]+),?.*/	[\2] = \"\1\",/g'
 printf "};\n"
diff -ruN '--exclude=.git' oasis-base/tools/perf/trace/beauty/sndrv_pcm_ioctl.sh oasis-linux/tools/perf/trace/beauty/sndrv_pcm_ioctl.sh
--- oasis-base/tools/perf/trace/beauty/sndrv_pcm_ioctl.sh	2026-04-12 20:48:06.000000000 +0000
+++ oasis-linux/tools/perf/trace/beauty/sndrv_pcm_ioctl.sh	2026-05-26 06:38:01.578039143 +0000
@@ -4,6 +4,6 @@
 [ $# -eq 1 ] && beauty_uapi_sound_dir=$1 || beauty_uapi_sound_dir=tools/perf/trace/beauty/include/uapi/sound/
 
 printf "static const char *sndrv_pcm_ioctl_cmds[] = {\n"
-grep "^#define[\t ]\+SNDRV_PCM_IOCTL_" $beauty_uapi_sound_dir/asound.h | \
-	sed -r 's/^#define +SNDRV_PCM_IOCTL_([A-Z0-9_]+)[\t ]+_IO[RW]*\( *.A., *(0x[[:xdigit:]]+),?.*/\t[\2] = \"\1\",/g'
+grep "^#define[	 ]\+SNDRV_PCM_IOCTL_" $beauty_uapi_sound_dir/asound.h | \
+	sed -E 's/^#define +SNDRV_PCM_IOCTL_([A-Z0-9_]+)[	 ]+_IO[RW]*\( *.A., *(0x[[:xdigit:]]+),?.*/	[\2] = \"\1\",/g'
 printf "};\n"
diff -ruN '--exclude=.git' oasis-base/tools/power/x86/intel-speed-select/Makefile oasis-linux/tools/power/x86/intel-speed-select/Makefile
--- oasis-base/tools/power/x86/intel-speed-select/Makefile	2026-04-12 20:48:06.000000000 +0000
+++ oasis-linux/tools/power/x86/intel-speed-select/Makefile	2026-05-26 06:38:01.621482528 +0000
@@ -53,7 +53,7 @@
 clean:
 	rm -f $(ALL_PROGRAMS)
 	rm -rf $(OUTPUT)include/linux/isst_if.h
-	find $(or $(OUTPUT),.) -name '*.o' -delete -o -name '\.*.d' -delete
+	find $(or $(OUTPUT),.) '(' -name '*.o' -o -name '\.*.d' ')' -exec $(RM) {} +
 
 install: $(ALL_PROGRAMS)
 	install -d -m 755 $(DESTDIR)$(bindir);		\
diff -ruN '--exclude=.git' oasis-base/tools/spi/Makefile oasis-linux/tools/spi/Makefile
--- oasis-base/tools/spi/Makefile	2026-04-12 20:48:06.000000000 +0000
+++ oasis-linux/tools/spi/Makefile	2026-05-26 06:38:01.626783266 +0000
@@ -53,9 +53,7 @@
 clean:
 	rm -f $(ALL_PROGRAMS)
 	rm -rf $(OUTPUT)include/
-	find $(or $(OUTPUT),.) -name '*.o' -delete
-	find $(or $(OUTPUT),.) -name '\.*.o.d' -delete
-	find $(or $(OUTPUT),.) -name '\.*.o.cmd' -delete
+	find $(or $(OUTPUT),.) '(' -name '*.o' -o -name '\.*.o.d' -o -name '\.*.o.cmd' ')' -exec $(RM) {} +
 
 install: $(ALL_PROGRAMS)
 	install -d -m 755 $(DESTDIR)$(bindir);		\
diff -ruN '--exclude=.git' oasis-base/tools/usb/Makefile oasis-linux/tools/usb/Makefile
--- oasis-base/tools/usb/Makefile	2026-04-12 20:48:06.000000000 +0000
+++ oasis-linux/tools/usb/Makefile	2026-05-26 06:38:01.878656919 +0000
@@ -38,7 +38,7 @@
 
 clean:
 	rm -f $(ALL_PROGRAMS)
-	find $(or $(OUTPUT),.) -name '*.o' -delete -o -name '\.*.d' -delete -o -name '\.*.o.cmd' -delete
+	find $(or $(OUTPUT),.) '(' -name '*.o' -o -name '\.*.d' -o -name '\.*.o.cmd' ')' -exec $(RM) {} +
 
 install: $(ALL_PROGRAMS)
 	install -d -m 755 $(DESTDIR)$(bindir);		\
